### PR TITLE
Tidy Image API language

### LINKED ISSUE #1425: Tidy language in Image API 3.0
We have a number of inconsistent uses of language in the Image API spec that should be tidied, things to look at include:

  * image / region / extracted region
  * image / image content
  * determined / defined / requested/ specificed

### PR DESCRIPTION
**Pull request by zimeon**
Fixes #1425

### CODE REVIEW

File: source/api/image/3.0/index.md [hunk 1]
```diff
@@ -136,25 +136,27 @@ The API places no restrictions on the form of the identifiers that a server may
 
 ##  4. Image Requests
 
-All parameters described below are required for compliant construction of a IIIF Image API URI. The sequence of parameters in the URI _MUST_ be in the order described below. The order of the parameters is also intended as a mnemonic for the order of the operations by which the service should manipulate the image content. Thus, the requested image content is first extracted as a region of the complete image, then scaled to the requested size, mirrored and/or rotated, and finally transformed into the color quality and format. This resulting image content is returned as the representation for the URI. 
+This specification makes no assumptions about the form of the underlying image content. The term **full image** is used to refer to the entire area of the underlying image content, with the pixel dimensions given in the [Image Information][image-information], and which is imagined as the starting point for image requests.
```
Comment (jpstroop) [inline source/api/image/3.0/index.md]: > the form of the underlying image content.

Strengthen this with examples, e.g. the content might be derived from a source image, but may also be dynamically generated ...

File: source/api/image/3.0/index.md [hunk 2]
```diff
@@ -136,25 +136,27 @@ The API places no restrictions on the form of the identifiers that a server may
...
+All parameters described below are required for compliant construction of a IIIF Image API URI. The sequence of parameters in the URI _MUST_ be in the order described below. The order of the parameters is also intended as a mnemonic for the order of the operations by which the service should manipulate the image content. Thus, the requested image content is first extracted as a region of the full image, then scaled to the requested size, mirrored and/or rotated, and finally transformed into the requested color quality and format. This resulting image is returned as the representation for the URI.
 
-Servers should support [CORS][cors-response] on Image responses.
+Size and region parameters in pixels _MUST_ be non-negative integers. Size and region parameters in percentages and the rotation parameter _MUST_ be positive floating point numbers or integers. These should have at most 10 decimal digits and consist only of decimal digits and "." with a leading zero if less than 1.0. Intermediate calculations may use floating point numbers and the rounding method is implementation specific.
```
Comment (jpstroop) [inline source/api/image/3.0/index.md]: > These should have at most 10 decimal digits

Do we really care? Not really related to this PR, just calling it out.

File: source/api/image/3.0/index.md [hunk 3]
```diff
@@ -136,25 +136,27 @@ The API places no restrictions on the form of the identifiers that a server may
...
+
+Servers should support [CORS][cors-response] on image responses.
```
Comment (jpstroop) [inline source/api/image/3.0/index.md]: _SHOULD_ ?

File: source/api/image/3.0/index.md [hunk 4]
```diff
@@ -276,7 +279,7 @@ The rotation parameter specifies mirroring and rotation. A leading exclamation m
 
 A rotation value that is out of range or unsupported _SHOULD_ result in a 400 (Bad Request) status code.
 
-In most cases a rotation will change the width and height dimensions of the returned image. The service _SHOULD_ return an image that contains all of the image contents requested in the region and size parameters, even if the dimensions of the returned image file are different than specified in the size parameter. The image contents _SHOULD NOT_ be scaled as a result of the rotation, and there _SHOULD_ be no additional space between the corners of the rotated image contents and the bounding box of the returned image content.
+In most cases a rotation will change the width and height dimensions of the returned image. The service _SHOULD_ return an image that contains all of the image contents requested in the region and size parameters, even if the dimensions of the returned image file are different than specified in the size parameter. The image contents _SHOULD NOT_ be scaled as a result of the rotation, and there _SHOULD_ be no additional space between the corners of the rotated image contents and the bounding box of the returned image.
```
Comment (jpstroop) [inline source/api/image/3.0/index.md]: > In most cases a rotation will...

In most cases, rotation will...

File: source/api/image/3.0/index.md [hunk 5]
```diff
@@ -424,15 +427,15 @@ It is possible to request the same image using different combinations of paramet
   * Caching becomes significantly more efficient, both client and server side, when the URIs used are the same between systems and sessions.
   * Response times can be improved by avoiding redirects from a requested non-canonical URI syntax to the canonical syntax by using the canonical form directly.
 
-In order to support the above requirements, clients _SHOULD_ construct the image request URIs using the following canonical parameter values where possible.  Image servers _MAY_ redirect the client to the canonical URI from a non-canonical equivalent.
+In order to support the above requirements, clients _SHOULD_ construct image request URIs using the following canonical parameter values where possible.  Image servers _MAY_ redirect the client to the canonical URI from a non-canonical equivalent.
 
 | Parameter | Canonical value |
 | --------- | --------------- |
-| region    | "full" if the whole image is requested<br/>otherwise the `x,y,w,h` syntax. |
+| region    | "full" if the full image is requested<br/>otherwise the `x,y,w,h` syntax. |
```
Comment (jpstroop) [inline source/api/image/3.0/index.md]: Same as https://github.com/IIIF/api/pull/1462/files#r170224974

File: source/api/image/3.0/index.md [hunk 6]
```diff
@@ -486,14 +489,16 @@ The JSON response has several technical properties that describe the available f
 | `type` | Required | The type for the Image API. The value _MUST_ be the string `ImageService3`. |
 | `protocol` | Required | The URI `http://iiif.io/api/image` which can be used to determine that the document describes an image service which is a version of the IIIF Image API. |
 | `profile` | Required | A string indicating the highest [compliance level][compliance-levels] which is fully supported by the service. The value _MUST_ be one of "level0", "level1", or "level2". |
-| `width` | Required | The width in pixels of the full image content, given as an integer. |
-| `height` | Required | The height in pixels of the full image content, given as an integer. |
+| `width` | Required | The width in pixels of the full image, given as an integer. |
+| `height` | Required | The height in pixels of the full image, given as an integer. |
```
Comment (jpstroop) [inline source/api/image/3.0/index.md]: Don't you want "image content" here?

File: source/api/image/3.0/index.md [hunk 7]
```diff
@@ -486,14 +489,16 @@ The JSON response has several technical properties that describe the available f
...
+The `width` and `height` properties give the size of the full image and are required in order to construct tile requests.
+
+The `maxWidth`, `maxHeight` and `maxArea` parameters provide a way for image servers to express limits on the sizes supported for the image. If `maxWidth` alone, or `maxWidth` and `maxHeight` are specified then clients should expect requests with larger linear dimensions to be rejected. If `maxArea` is specified then clients should expect requests with larger pixel areas to be rejected. The `maxWidth / maxHeight`  and `maxArea` parameters are independent, servers may implement either or both limits. Servers _MUST_ ensure that sizes specified by any `sizes` or `tiles` properties are within any size limits expressed. Clients _SHOULD NOT_ make requests that exceed size limits expressed.
```
Comment (jpstroop) [inline source/api/image/3.0/index.md]: > The `maxWidth`, `maxHeight` and `maxArea`

Serial comma...

File: source/api/image/3.0/index.md (hunk 1 above)
Comment (tomcrane) [reply source/api/image/3.0/index.md]: "...imagined as..." -> considered as?

File: source/api/image/3.0/index.md (hunk 2 above)
Comment (zimeon) [reply source/api/image/3.0/index.md]: -> #1468

File: source/api/image/3.0/index.md (hunk 1 above)
Comment (zimeon) [reply source/api/image/3.0/index.md]: Happy to add examples, would prefer to keep "imagined" over "considered"

File: source/api/image/3.0/index.md [hunk 8]
```diff
@@ -136,25 +136,27 @@ The API places no restrictions on the form of the identifiers that a server may
...
 ### 4.1. Region
 
-The region parameter defines the rectangular portion of the complete image to be returned. Region can be specified by pixel coordinates, percentage or by the value "full", which specifies that the entire image should be returned.
+The region parameter defines the rectangular portion of the underlying image content to be returned. Region can be specified by pixel coordinates, percentage or by the value "full", which specifies that the full image should be returned.
 
 | Form |  Description |
 | ------------------- | ------------ |
-| `full`              | The complete image is returned, without any cropping. |
-| `square`            | The region is defined as an area where the width and height are both equal to the length of the shorter dimension of the complete image. The region may be positioned anywhere in the longer dimension of the image content at the server's discretion, and centered is often a reasonable default.|
+| `full`              | The full image is returned, without any cropping. |
```
[the remark this answers is not in the document]
Comment (zimeon) [reply source/api/image/3.0/index.md]: I think we are in the sense that **full image** is the starting point of the region selection, so `full` means that we keep all of that and return the full image

File: source/api/image/3.0/index.md (hunk 5 above)
Comment (zimeon) [reply source/api/image/3.0/index.md]: As in https://github.com/IIIF/api/pull/1462/files#r170224974, I think it should be **full image** which we define as the starting point for the transformations

File: source/api/image/3.0/index.md (hunk 7 above)
Comment (zimeon) [reply source/api/image/3.0/index.md]: ... is especially bad in cases with a short list of single words

Comment (zimeon) [reply source/api/image/3.0/index.md]: Perhaps we need a style guide decision on this one

**Comment by jpstroop** [comment]
I guess what’s weird is that something that theoretically doesn’t exist has a `width` and `height`.

Arguably, `width` and `height` are properties of the service, not the image

**Pull request merged**